src/mailchimpRoot.php

```diff
@@ -339,6 +339,20 @@ public function DELETE()
 
     // BEGIN OAUTH FUNCTIONS
 
+    public static function getAuthUrl(
+        $client_id,
+        $redirect_uri
+    ){
+        $encoded_uri = urldecode($redirect_uri);
+
+        $authUrl = "https://login.mailchimp.com/oauth2/authorize";
+        $authUrl .= "?client_id=".$client_id;
+        $authUrl .= "&redirect_uri=".$encoded_uri;
+        $authUrl .= "&response_type=code";
```

Comment:
Not sure if you knew this existed. But PHP has this that can be used in places like this: http://php.net/manual/en/function.http-build-query.php

Reply:
I know, I just wanted my code to match yours in oauthExchange(). Not sure if it is what you'd like.

Reply:
This is cool, I just wanted to share that, just in case. 🐢  I'm going to go ahead and merge this! Thanks again, and feel free to open other PR's.